src/NUnitFramework/tests/CakeTests.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Charlie Poole, Rob Prouse and Contributors. MIT License - see LICENSE.txt
+
+namespace NUnit.Framework.Tests;
+
+public class CakeTests
+{
+    [TestCase("1.2.3-beta.1", "1.2.3.1")]
+    [TestCase("1.2.3", "1.2.3.0")]
+    public void ThatWeCanParseFileVersionStrings(string versionString, string expected)
+    {
+        var cake = new CakeCode();
+        var parsedVersion = cake.ParseFileVersion(versionString);
+        Assert.That(parsedVersion, Is.EqualTo(expected));
+    }
+
+    [TestCase("1.2.3-beta.1", "1.2.3.0")]
+    [TestCase("1.2.3", "1.2.3.0")]
+    public void ThatWeCanParseAssemblyVersionStrings(string versionString, string expected)
+    {
+        var cake = new CakeCode();
+        var parsedVersion = cake.ParseAssemblyVersion(versionString);
+        Assert.That(parsedVersion, Is.EqualTo(expected));
+    }
+}
+
+/// <summary>
+/// This class is copied into the cake script. It is just here to be tested.
```

Comment:
How do we keep this in sync so we are testing what is used?

Reply:
I looked up how to include source in cake, but didnt see how to go all the way there.  Just wanted now to make sure the code was tested - to avoid missing something.  But we should include the code.  I can ask the Cake people later.

Reply:
Done